### PR TITLE
fix: split rust-std build into unix and win version

### PR DESCRIPTION
**Pull request by ruben-arts**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged) **By design the build strings will be changed.**
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

PR as a fix for the talk in #153

### CODE REVIEW

**Comment by ruben-arts** [comment]
The CI is succeeding but I'm not sure that it works as I intended, could you check the work @pkgw @isuruf ?

File: recipe/install-rust-std-extra.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+echo %PKG_NAME% > ./components
+
+bash -lce "install.sh --prefix="%PREFIX%" --destdir="%DESTDIR%""
+
+del "%PREFIX%/lib/rustlib/components"
+del "%PREFIX%/lib/rustlib/install.log"
+del "%PREFIX%/lib/rustlib/rust-installer-version"
+del "%PREFIX%/lib/rustlib/uninstall.sh"
+
+endlocal
```
Comment (isuruf) [inline recipe/install-rust-std-extra.bat:15]: Can you use the contents of `recipe/install-rust-std.bat` here with `s/install-rust-std.sh/install-rust-std-extra.sh/g`

**Review by isuruf — CHANGES_REQUESTED** [review]
You need to change lines 30-56 to change the selector from `linux and x86_64` to `(linux or win) and x86_64`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -163,20 +163,32 @@ outputs:
...
+        - __unix  # [unix]
       run_constrained:
         # Having different versions of rust-std and rust is confusing.
         - {{ pin_subpackage("rust", min_pin="x.x.x", max_pin="x.x.x") }}
     test:
       commands:
-        - test -d $PREFIX/lib/rustlib/{{ rust_std_extra }}
-        - test -f $PREFIX/lib/rustlib/manifest-rust-std-{{ rust_std_extra }}
+        - test -d $PREFIX/lib/rustlib/{{ rust_std_extra }}  # [unix]
+        - if not exists "%LIBRARY_PREFIX%/lib/rustlib/{{ rust_std_extra }}" exit 1  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:186]:
```suggestion
        - if not exist "%LIBRARY_PREFIX%/lib/rustlib/{{ rust_std_extra }}" exit 1  # [win]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -163,20 +163,32 @@ outputs:
...
+        - test -f $PREFIX/lib/rustlib/manifest-rust-std-{{ rust_std_extra }}  # [unix]
+        - if not exists "%LIBRARY_PREFIX%/lib/rustlib/manifest-rust-std-{{ rust_std_extra }}" exit 1  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:188]:
```suggestion
        - if not exist "%LIBRARY_PREFIX%/lib/rustlib/manifest-rust-std-{{ rust_std_extra }}" exit 1  # [win]
```

**Pull request merged**